File: Level1/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    int pre = arr[0];
    answer.push_back(pre);
    
    for(int i = 1; i < arr.size(); i++)
    {
        if(pre != arr[i])
        {
            pre = arr[i];
            answer.push_back(pre);
        }
    }
    
    return answer;
}

File: Level1/같은 숫자는 싫어.js
function solution(arr)
{
    let answer = [];
    let pre = arr[0];
    
    answer.push(pre);
    
    for(let i = 1; i < arr.length; i++){
        if(arr[i] !== pre){
            answer.push(arr[i]);
            pre = arr[i];
        }
    }
    
    return answer;
    //return arr.filter((val,index) => val != arr[index+1]);
}